src/components/Modals/Receive/Receive.tsx

```diff
@@ -11,97 +11,153 @@ import {
   ModalFooter,
   ModalHeader,
   ModalOverlay,
-  useColorModeValue
+  useColorModeValue,
+  useToast
 } from '@chakra-ui/react'
+import { ChainIdentifier } from '@shapeshiftoss/chain-adapters'
+import { AssetMarketData } from '@shapeshiftoss/market-service'
 import { Card } from 'components/Card'
 import { QRCode } from 'components/QRCode/QRCode'
 import { RawText, Text } from 'components/Text'
+import { useChainAdapters } from 'context/ChainAdaptersProvider/ChainAdaptersProvider'
 import { useModal } from 'context/ModalProvider/ModalProvider'
+import { useWallet } from 'context/WalletProvider/WalletProvider'
+import { useEffect, useState } from 'react'
 import { useTranslate } from 'react-polyglot'
 
-const Receive = () => {
+type ReceivePropsType = {
+  asset: AssetMarketData
+}
+
+const Receive = ({ asset }: ReceivePropsType) => {
+  const { name, symbol } = asset
+  const { state } = useWallet()
+  const [isNativeWallet, setIsNativeWallet] = useState<boolean>(true)
+  const [receiveAddress, setReceiveAddress] = useState<string>('')
+  const chainAdapterManager = useChainAdapters()
+
+  useEffect(() => {
+    ;(async () => {
+      const { wallet } = state
+      if (!wallet) return
+      setIsNativeWallet((await wallet.getLabel()) === 'Native')
+      const chainAdapter = chainAdapterManager.byChain(ChainIdentifier.Ethereum)
+      // TODO(0xdef1cafe): remove this when chain adapters has a default path
+      const path = `m/44'/60'/0'/0/0`
+      setReceiveAddress(await chainAdapter.getAddress({ wallet, path }))
+    })()
+  }, [chainAdapterManager, state, setReceiveAddress])
+
   const translate = useTranslate()
-  const modal = useModal()
+  const { receive } = useModal()
+  const toast = useToast()
+  const { close, isOpen } = receive
+
+  const hoverColor = useColorModeValue('gray.900', 'white')
+  const bg = useColorModeValue('gray.100', 'gray.700')
+
+  const copyHandler = () => {
+    navigator.clipboard.writeText(receiveAddress)
+    toast({
+      title: translate('modals.receive.copied', { symbol: symbol.toUpperCase() }),
+      description: receiveAddress,
+      status: 'success',
+      duration: 2500,
+      isClosable: true
+    })
+  }
+
+  const Verify = () => (
+    <Button
+      color='gray.500'
+      flexDir='column'
+      role='group'
+      variant='link'
+      _hover={{ textDecoration: 'none', color: hoverColor }}
+      onClick={() =>
+        toast({
+          title: 'unimplemented',
+          description: 'unimplemented',
+          status: 'error',
+          duration: 2500,
+          isClosable: true
+        })
+      }
+    >
+      <Circle bg={bg} mb={2} size='40px' _groupHover={{ bg: 'blue.500', color: 'white' }}>
+        <ViewIcon />
+      </Circle>
+      <Text translation='modals.receive.verify' />
+    </Button>
+  )
+
+  const ConnectAWallet = () => (
```

Comment:
Already spoke in person, but we should use the AssetActions to trigger the connect wallet modal as done here vs letting this logic live in the receive modal:

https://github.com/shapeshift/web/pull/33/files#diff-bac7e67d6a6928ba7679eea90ee123bb214b4261c234f554cfe55f092623983a

Reply:
ty ser